oauth2-authorization-server/src/main/java/org/springframework/security/oauth2/server/authorization/authentication/OAuth2AuthorizationCodeRequestAuthenticationProvider.java

```diff
@@ -124,17 +124,26 @@ public Authentication authenticate(Authentication authentication) throws Authent
 					authorizationCodeRequestAuthentication, registeredClient);
 		}
 
+		OAuth2AuthorizationRequest authorizationRequest = OAuth2AuthorizationRequest.authorizationCode()
```

Comment:
Please revert all changes and instead add parameter `OAuth2AuthorizationCodeRequestAuthenticationToken authorizationCodeRequestAuthentication` to `resolveRedirectUri()` allowing usage of `authorizationCodeRequestAuthentication.redirectUri`.